web3.js/src/address-lookup-table-program.ts

```diff
@@ -0,0 +1,295 @@
+import {toBufferLE} from 'bigint-buffer';
+import * as BufferLayout from '@solana/buffer-layout';
+
+import * as Layout from './layout';
+import {PublicKey} from './publickey';
+import * as bigintLayout from './util/bigint';
+import {SystemProgram} from './system-program';
+import {TransactionInstruction} from './transaction';
+import {encodeData, IInstructionInputData} from './instruction';
+
+export type CreateLookupTableParams = {
+  /** Account used to derive and control the new address lookup table. */
+  authority: PublicKey;
+  /** Account that will fund the new address lookup table. */
+  payer: PublicKey;
+  /** A recent slot must be used in the derivation path for each initialized table. */
+  recentSlot: bigint;
+};
+
+export type FreezeLookupTableParams = {
+  /** Address lookup table account to freeze. */
+  lookupTable: PublicKey;
+  /** Account which is the current authority. */
+  authority: PublicKey;
+};
+
+export type ExtendLookupTableParams = {
+  /** Address lookup table account to freeze. */
+  lookupTable: PublicKey;
+  /** Account which is the current authority. */
+  authority: PublicKey;
+  /** Account that will fund the table reallocation. */
+  payer?: PublicKey;
+  /** List of Public Keys to be added to the lookup table. */
+  addresses: Array<PublicKey>;
+};
+
+export type DeactivateLookupTableParams = {
+  /** Address lookup table account to freeze. */
+  lookupTable: PublicKey;
+  /** Account which is the current authority. */
+  authority: PublicKey;
+};
+
+export type CloseLookupTableParams = {
+  /** Address lookup table account to freeze. */
+  lookupTable: PublicKey;
+  /** Account which is the current authority. */
+  authority: PublicKey;
+  /** Recipient of closed account lamports. */
+  recipient: PublicKey;
+};
+
+export type LookupTableInstructionType =
+  | 'CreateLookupTable'
+  | 'ExtendLookupTable'
+  | 'CloseLookupTable'
+  | 'FreezeLookupTable'
+  | 'DeactivateLookupTable';
+
+export type LookupTableInstructionInputData = {
+  CreateLookupTable: IInstructionInputData &
+    Readonly<{
+      recentSlot: bigint;
+      bumpSeed: number;
+    }>;
+  FreezeLookupTable: IInstructionInputData;
+  ExtendLookupTable: IInstructionInputData &
+    Readonly<{
+      numberOfAddresses: bigint;
+      addresses: Array<Uint8Array>;
+    }>;
+  DeactivateLookupTable: IInstructionInputData;
+  CloseLookupTable: IInstructionInputData;
+};
+
+export const LOOKUP_TABLE_INSTRUCTION_LAYOUTS = Object.freeze({
+  Create: {
```

Comment:
Can you add the `LookupTable` suffix to these key names as well?

Reply:
Fixed.